src/contracts/collections/pools.rs

```diff
@@ -45,4 +37,93 @@ impl Pools {
         self.pools.remove(&pool_key);
         Ok(())
     }
+
+    pub fn get(&self, pool_key: PoolKey) -> Option<Pool> {
```

Comment:
This looks inconsistent as in other collections the `Result` is returned instead of option